src/reconcile/rec_visibility.c

```diff
@@ -132,6 +133,13 @@ __rec_append_orig_value(
             tombstone->next = append;
             total_size += size;
         }
+        /*
+         * Assert that if we did have a tombstone as the original update that its start timestamp
+         * matches the previous updates stop timestamp. Note that the stop_ts on the previous update
+         * can be WT_TS_MAX so we should ignore this.
+         */
+        WT_ASSERT(session, upd->type != WT_UPDATE_TOMBSTONE || unpack->stop_ts == WT_TS_MAX ||
+            unpack->stop_ts == upd->start_ts);
```

Comment:
Why not assert transaction id matches? In this case, we shouldn't clear the transaction ids so it should match.

Reply:
The transaction IDs don't match if you look at the core I put in WT-5655, unless I'm missing something.

Reply:
May be something like this?
#ifdef HAVE_DIAGNOSTICS
else if (upd ->type == WT_UPDATE_TOMBSTONE)
      WT_ASSERT(session, unpack->stop_txn == upd->txnid && unpack->start_ts == upd->start_ts);
#5247